### ISSUE TITLE
[HTML2] Curl example copies HTTP verb(GET/POST) in lower case in Firefox

### ISSUE THREAD

**Issue by svarogg** [swagger-api/swagger-codegen#5558]
##### Description

The curl example shows i.e. `curl -X GET "http://localhost:8000/greetings/hello/{subject}"`.
However, if you copy it in firefox (In chromium it does not reproduce) you get `curl -X get "http://localhost:8000/greetings/hello/{subject}"` which fails if run from the command line.

It seems like the verb is capitalized through `text-transform: uppercase;` which does not get represented when copied from firefox.

##### Swagger-codegen version
v2.2.2

##### Swagger declaration file content or url
This bug reproduces for any swagger file.
For example, you can use the hello-world sample from https://github.com/OAI/OpenAPI-Specification/wiki/Hello-World-Sample

##### Command line used for generation
java -jar /tmp/swagger-codegen-cli.jar generate -i swagger.json -l html2

##### Steps to reproduce

1. Create any json or yaml swagger specification
2. Run the above command line to generate an index.html file
3. Open said html file with Firefox
4. Copy the curl sample to the clipboard, paste in terminal and run.

##### Related issues
Firefox bug to fix this issue upstream:
https://bugzilla.mozilla.org/show_bug.cgi?id=35148
Doesn't look like they are planning to fix it anytime soon.

##### Suggest a Fix
Actually capitalize the http-verb instead of just the way it's displayed.
It seems like this is not possible through mustache, so maybe use a small javascript snippet to do that.

**Comment by wing328** [comment]
@svarogg I agree that we should do it in mustache instead of with Javascript.

May I know if you've time to contribute the fix (which should be pretty straightforward)? I can show you some good starting points.

**Comment by svarogg** [comment]
I've began looking into a solution...
It seems like the problem is in swagger-codegen/src/main/resources/htmlDocs2/index.mustache line 227., specifically: `<span style="text-transform: uppercase;">{{httpMethod}}</span>`

It would be nice if we could tell mustache to capitalize the thing, but apparently it's not possible by-design (See http://stackoverflow.com/questions/8460720/mustache-jquery-javascript-how-to-perform-method-on-mustache-variable).

So, unless I misunderstand something, we have two options I can think of:
1) Capitalize with Javascript. Would work, but very ugly.
2) Capitalize the source. I guess the source goes to all other templates, and maybe used in places where capitalization is not desirable.
A solution could be the introduction of another value (say httpMethodCapitalized or something).

I can try fixing it myself if I'm pointed towards the desired solution and (for solution No. 2) the places in code that generate the values that go into mustache.

**Comment by wing328** [comment]
@svarogg we can apply solution similar to https://github.com/swagger-api/swagger-codegen/blob/7fc8b2400149d21d36663b310651665f42b11de5/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/TypeScriptAngular2ClientCodegen.java#L175-L212 by overriding postProcessOperations and then updating `httpMethod` to upper case.

**Comment by svarogg** [comment]
I see that currently  it's converted to lower-case specifically:
https://github.com/swagger-api/swagger-codegen/blame/7fc8b2400149d21d36663b310651665f42b11de5/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/StaticHtml2Generator.java#L99
Is there any reason for this?

If not I guess I can flip it to .toUpperCase and see what it does.

**Comment by wing328** [comment]
```
swagger-codegen|master⚡ ⇒ grep -R httpMethod modules/swagger-codegen/src/main/resources/htmlDocs2/*
modules/swagger-codegen/src/main/resources/htmlDocs2/index.mustache:                        <pre class="prettyprint language-html prettyprinted" data-type="{{httpMethod}}"><code><span class="pln">{{path}}</span></code></pre>
modules/swagger-codegen/src/main/resources/htmlDocs2/index.mustache:                            <pre class="prettyprint"><code class="language-bsh">curl -X <span style="text-transform: uppercase;">{{httpMethod}}</span>{{#authMethods}}{{#isApiKey}}{{#isKeyInHeader}} -H "{{keyParamName}}: [[apiKey]]"{{/isKeyInHeader}}{{/isApiKey}}{{#isBasic}}{{#hasProduces}} -H "Accept: {{#produces}}{{{mediaType}}}{{#hasMore}},{{/hasMore}}{{/produces}}"{{/hasProduces}}{{#hasConsumes}} -H "Content-Type: {{#consumes}}{{{mediaType}}}{{#hasMore}},{{/hasMore}}{{/consumes}}"{{/hasConsumes}} -H "Authorization: Basic [[basicHash]]"{{/isBasic}}{{/authMethods}} "{{basePath}}{{path}}{{#hasQueryParams}}?{{#queryParams}}{{^-first}}&{{/-first}}{{baseName}}={{vendorExtensions.x-eg}}{{/queryParams}}{{/hasQueryParams}}"</code></pre>
```

I did a grep and found that `httpMethod` is used in more than 1 place.

I suggest we leverage vendor extension (e.g. x-codegen-httpMethodUpperCase) similar to https://github.com/swagger-api/swagger-codegen/blame/7fc8b2400149d21d36663b310651665f42b11de5/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/StaticHtml2Generator.java#L130 to keep the original httpMethod in lower case.

**Comment by wing328** [comment]
@svarogg let me know if you need help with the fix/PR.

**Comment by wing328** [comment]
Closed via #5705
